contracts/EntityCoreFacet.sol

```diff
@@ -14,175 +15,192 @@ import "./Policy.sol";
-    stakeholders[2] = _stakeholders[0];
-    stakeholders[3] = _stakeholders[1];
-    stakeholders[4] = _stakeholders[2];
-    stakeholders[5] = _stakeholders[3];
-
-    require(
-      IAccessControl(stakeholders[2]).aclContext() == aclContext(), 
-      'underwriter ACL context must match'
-    );
-
-    Policy f = new Policy(
-      address(settings()),
-      stakeholders,
-      _dates,
-      _unit,
-      _premiumIntervalSeconds,
-      _commmissionsBP
-    );
-
-    uint256 numPolicies = dataUint256["numPolicies"];
-    address pAddr = address(f);
-    dataAddress[__i(numPolicies, "policy")] = pAddr;
-    dataUint256["numPolicies"] = numPolicies + 1;
-    dataBool[__a(pAddr, "isPolicy")] = true; // for _isPolicyCreatedByMe() to work
-
-    emit NewPolicy(pAddr, address(this), msg.sender);
-  }
-
-  function getBalance(address _unit) public view override returns (uint256) {
-    return dataUint256[__a(_unit, "balance")];
-  }
-
-  function getNumPolicies() public view override returns (uint256) {
-    return dataUint256["numPolicies"];
-  }
-
-  function getPolicy(uint256 _index) public view override returns (address) {
-    return dataAddress[__i(_index, "policy")];
-  }
-
-  function deposit(address _unit, uint256 _amount) 
-    external 
-    override 
-  {
-    IERC20 tok = IERC20(_unit);
-    tok.transferFrom(msg.sender, address(this), _amount);
-    dataUint256[__a(_unit, "balance")] = dataUint256[__a(_unit, "balance")].add(_amount);
-    
-    emit EntityDeposit(msg.sender, _unit, _amount);
-  }
-
-  function withdraw(address _unit, uint256 _amount) 
-    external 
-    override 
-    assertIsEntityAdmin(msg.sender)
-  {
-    _assertHasEnoughBalance(_unit, _amount);
-
-    dataUint256[__a(_unit, "balance")] = dataUint256[__a(_unit, "balance")].sub(_amount);
-
-    IERC20 tok = IERC20(_unit);
-    tok.transfer(msg.sender, _amount);
-
-    emit EntityWithdraw(msg.sender, _unit, _amount);
-  }
-
-  function payTranchPremium(address _policy, uint256 _tranchIndex, uint256 _amount)
-    external
-    override
-    assertCanPayTranchPremiums(_policy)
-  {
-    address policyUnitAddress;
-
-    IPolicy p = IPolicy(_policy);
-
-    // avoid stack too deep errors
+contract EntityCoreFacet is
+    EternalStorage,
+    Controller,
+    EntityFacetBase,
+    IEntityCoreFacet,
+    IDiamondFacet
+{
+    using SafeMath for uint256;
+
+    modifier assertCanTradeTranchTokens() {
+        require(inRoleGroup(msg.sender, ROLEGROUP_TRADERS), "must be trader");
+        _;
+    }
+
+    modifier assertCanPayTranchPremiums(address _policyAddress) {
+        require(
+            inRoleGroup(msg.sender, ROLEGROUP_ENTITY_REPS),
+            "must be entity rep"
+        );
+        _;
+    }
+
+    /**
+     * Constructor
+     */
+    constructor(address _settings) public Controller(_settings) {}
+
+    // IDiamondFacet
+    function getSelectors() public pure override returns (bytes memory) {
+        return
+            abi.encodePacked(
```

Comment:
These changes are just about formatting and indentation. Pls fix to use 2 spaces for indents everywhere.